code/backend/batch/BatchPushResults.py

```diff
@@ -39,24 +37,11 @@ def do_batch_push_results(msg: func.QueueMessage) -> None:
         msg.get_body().decode("utf-8"),
     )
 
-    document_processor = DocumentProcessor()
     blob_client = AzureBlobStorageClient()
     # Get the file name from the message
     file_name = _get_file_name_from_message(msg)
     # Generate the SAS URL for the file
     file_sas = blob_client.get_blob_sas(file_name)
-    # Get file extension's processors
-    file_extension = file_name.split(".")[-1]
-
     # Process the file
-    if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
-        document_processor.process_using_integrated_vectorisation(source_url=file_sas)
-    else:
-        processors = list(
-            filter(
-                lambda x: x.document_type.lower() == file_extension.lower(),
-                ConfigHelper.get_active_config_or_default().document_processors,
-            )
-        )
-        document_processor.process(source_url=file_sas, processors=processors)
-        blob_client.upsert_blob_metadata(file_name, {"embeddings_added": "true"})
+    embedder_factory = EmbedderFactory.create(env_helper)
+    embedder_factory.embed_file(file_sas, file_name)
```

Comment:
nit:
```suggestion
    embedder = EmbedderFactory.create(env_helper)
    embedder.embed_file(file_sas, file_name)
```

(The create creates an object - it doesnt retrun the factory)